packetbeat/_meta/fields.yml

```diff
@@ -1748,3 +1748,293 @@
             If the call resulted in exceptions, this field contains the exceptions in a human
             readable format.
 
+- key: tls
+  title: "TLS"
+  description: >
+    TLS-specific event fields.
+  fields:
+    - name: tls
+      type: group
+      fields:
+        - name: handshake_completed
+          type: boolean
+          description: >
+            Whether the TLS negotiation has been successful and the session has
+            transitioned to encrypted mode.
+
+        - name: resumed
+          type: boolean
+          description: >
+            If the TLS session has been resumed from a previous session.
+
+        - name: resumption_method
+          type: keyword
+          description: >
+            If the session has been resumed, the underlying method used. One of
+            "id" for TLS session ID or "ticket" for TLS ticket extension.
+
+        - name: client_certificate_requested
+          type: boolean
+          description: >
+            Whether the server has requested the client to authenticate itself
+            using a client certificate.
+
+        - name: client_hello
+          type: group
+          fields:
+            - name: version
+              type: keyword
+              description: >
+                The version of the TLS protocol by which the client wishes to
+                communicate during this session.
+
+            - name: timestamp
+              type: long
+              description: >
+                The current time and date in standard UNIX 32-bit format
+                according to the client's internal clock.
+
+            - name: supported_ciphers
+              type: array
+              description: >
+                List of ciphers the client is willing to use for this session.
+                See https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4
+
+            - name: supported_compression_methods
+              type: array
+              description: >
+                The list of compression methods the client supports.
+                See https://www.iana.org/assignments/comp-meth-ids/comp-meth-ids.xhtml
+
+            - name: extensions
+              type: group
+              description: The hello extensions provided by the client.
+              fields:
+                - name: server_name_indication
+                  type: string
+                  description: Comma-separated list of server names
```

Comment:
Instead of comma separated, I think a list would be better for searching and aggregating.